src/framework/uicomponents/qml/MuseScore/UiComponents/StyledPopupView.qml

```diff
@@ -29,20 +30,31 @@ PopupView {
     property bool opensUpward: false
     property var arrowX: width / 2
     property alias arrowHeight: arrow.height
+    property alias arrowVisible: arrow.visible
 
     property color borderColor: ui.theme.strokeColor
     property color fillColor: ui.theme.backgroundPrimaryColor
     readonly property int borderWidth: 1
 
+    function toggleOpened() {
+        if (isOpened) {
+            close()
+        } else {
+            open()
+        }
+    }
+
```

Comment:
Couldn't this be done in "PopupView.cpp"? Or has it advantages to do it in QML?

Reply:
Yes, you are right. Moved to cpp